src/timed_string.rs

```diff
@@ -0,0 +1,78 @@
+use std::time::{Duration, Instant};
+use bytes::Bytes;
+
+pub struct TimedString {
+    expires_at: Instant,
+    cache_for:  Duration,
+    url:  String,
+    pub text: Bytes,
+    pub updater: fn (&str) -> Bytes
+}
+
+impl TimedString {
+    pub fn new<S: Into<String>>(url: S, cache_for: Duration) -> TimedString {
```

Comment:
using `Into<String>` means we can accept both `&str` or `String` or anything else that converts to a string without requiring the caller to convert.